### ISSUE TITLE
considerable time spent in `setPrior` for IHT and BAR

### ISSUE THREAD

**Issue by egillax** [OHDSI/Cyclops#68]
Hi @msuchard ,

I'm experimenting with running a simple cross validation to select the `penalty` parameter for iterative hard thresholding and broken adaptive ridge. In the process I did some profiling to check if there were inefficiencies in my R code. Then I noticed that a lot of time seems to be spent in the function [`.cyclopsSetPrior`](https://github.com/OHDSI/Cyclops/blob/main/R/ModelFit.R#L188), considerably more than in the model fitting itself.

I then ran hotspot analysis with intel vtune which point to this [line](https://github.com/OHDSI/Cyclops/blob/main/src/cyclops/priors/JointPrior.h#L49) as the culprit:

```cpp
void addVarianceParameter(const VariancePtr& ptr) {
        if (std::find(variance.begin(), variance.end(), ptr) == variance.end()) {
            variance.push_back(ptr);
        }
    }
```
When developing a model using IHT or BAR in PLP 23-40% of the total time is spent in these lines. Removing the if statement provides a significant speedup without affecting the output (in my case on simulated data at least). Is the if statement necessary for models such as `BAR` and `IHT` ? 

Regards,
Egill

**Comment by msuchard** [comment]
Hi @egillax -- fantastic find!!! I agree that for some models we do not need to check for uniqueness.   Can we add an extra boolean argument to the function to short-cut the `find`?    In the patch, make sure to not to change the R API.

**Comment by egillax** [comment]
Hi @msuchard ,

I did dive into this a bit more. This only seems to affect the running time when models have low N (~ 1000) and high P (~ 30k). When I tested this on data with higher N I couldn't observe any speedup by skipping the uniqueness check, I assume because the model fitting itself takes bulk of the time.

I'm not sure if you want to optimize for the case of low N - which is very quick anyway - at the cost of more complexity to the code. If you do I can submit a PR. Otherwise we can close this issue.
